src/libraries/Microsoft.PowerFx.Core/Parser/ParseResult.cs

```diff
@@ -55,6 +58,32 @@ public class ParseResult : IOperationStatus
         // All the spans in the tokens are relative to this. 
         public string Text { get; }
 
+        /// <summary>
+        /// If string is too large, don't even attempt to lex or parse it. 
+        /// Just create a parse result directly encapsulating the error. 
+        /// </summary>
+        /// <param name="text"></param>
+        /// <param name="maxAllowed">Maximum number of characters allowed. </param>
+        /// <returns></returns>
+        internal static ParseResult ErrorTooLarge(string text, int maxAllowed)
+        {
+            Token tok = new ErrorToken(new Span(0, text.Length));
+
+            var errKey = Core.Localization.TexlStrings.ErrTextTooLarge;
+            var err = new TexlError(tok, DocumentErrorSeverity.Critical, errKey, maxAllowed, text.Length);
+
+            List<TexlError> errors = new List<TexlError>()
+            {
+                err
+            };
+
+            int id = 0;
```

Comment:
>int id = 0; [](http://example.com/codeflow?start=12&length=11)

I think you can remove this line and use "ref int id" in next line

Reply:
only for 'out'. 
ref needs to be defined before usage.